components/brave_wallet_ui/components/shared/segmented-control/segmented-control.tsx

```diff
@@ -20,20 +20,38 @@ import {
 
 interface Props {
   navOptions: NavOption[]
+  useHashes?: boolean
   width?: number
 }
 
 export const SegmentedControl = (props: Props) => {
-  const { navOptions, width } = props
+  const { navOptions, width, useHashes } = props
 
   // Routing
   const history = useHistory()
-  const { pathname: walletLocation } = useLocation()
+  const { pathname: walletLocation, hash } = useLocation()
 
   // Methods
-  const onClick = (navOption: NavOption) => {
+  const onClick = React.useCallback((navOption: NavOption) => {
+    if (navOption.hash && useHashes) {
+      history.push(`${walletLocation}${navOption.hash}`)
+      return
+    }
     history.push(navOption.route)
-  }
+  }, [useHashes])
+
+  const isSelected = React.useCallback(
+    (navOption: NavOption) => {
+      return useHashes
+        ? !hash
+          // Since hashes are not necessary
+          // we default the first option as selected,
+          // if there is no hash in the route location.
+          ? navOptions[0].id === navOption.id
+          : hash === navOption.hash
+        : walletLocation.includes(navOption.route)
+
+    }, [walletLocation, navOptions, useHashes, hash])
```

Comment:
let's use relative navigation to handle this instead
ie: `history.push('#accounts')`

Reply:
Updated this with you! :)